pydicom/tests/test_charset.py

```diff
@@ -404,3 +404,19 @@ def test_invalid_second_encoding(self):
                                 u"- using default encoding instead"):
             pydicom.charset.decode(
                 elem, ['ISO 2022 IR 100', 'ISO 2022 IR 146'])
+
+    def test_japanese_multi_byte_encoding(self):
+        """Test japanese multi byte strings are correctly encoded."""
+        file_path = get_charset_files('chrH32.dcm')[0]
+        ds = dcmread(file_path)
+        ds.decode()
+
+        original_string = ds.PatientName.original_string
```

Comment:
This will not work in Python 2, where `PatientNameUnicode` is used, which does not have `original_string` - thus the `hasattr` condition in the `test_charset_patient_names` test.
Actually I thought it a good idea to incorporate the test for the original string into the `test_charset_patient_names` test, but as I tried that, other multi-value tests (for Chinese and Korean) also failed. I don't know much about these encodings, so any insight would be helful...

Reply:
I added if clause to skip this test in Python2. And according to ci result, it seems that it works correctly.

Reply:
I met the same errors you said. I think these are caused by the handling tail escape sequence and PersonaName data structure. After this PR will be finished, I want to discuss them.